src/town_gui.cpp

```diff
@@ -1584,6 +1606,64 @@ struct BuildHouseWindow : public PickerWindow {
 		}
 	}
 
+	static std::string GetHouseYear(TimerGameCalendar::Year min_year, TimerGameCalendar::Year max_year)
+	{
+		if (min_year == CalendarTime::MIN_YEAR) {
+			if (max_year == CalendarTime::MAX_YEAR) {
+				return GetString(STR_HOUSE_PICKER_YEARS_ANY);
+			}
+			SetDParam(0, max_year);
+			return GetString(STR_HOUSE_PICKER_YEARS_UNTIL);
+		}
+		if (max_year == CalendarTime::MAX_YEAR) {
+			SetDParam(0, min_year);
+			return GetString(STR_HOUSE_PICKER_YEARS_FROM);
+		}
+		SetDParam(0, min_year);
+		SetDParam(1, max_year);
+		return GetString(STR_HOUSE_PICKER_YEARS);
+	}
+
+	static std::string GetHouseInformation(const HouseSpec *hs)
+	{
+		std::stringstream line;
+
+		SetDParam(0, GetHouseName(hs));
+		line << GetString(STR_HOUSE_PICKER_NAME);
+		line << "\n";
+
+		SetDParam(0, hs->population);
+		line << GetString(STR_HOUSE_PICKER_POPULATION);
+		line << "\n";
+
+		line << GetHouseYear(hs->min_year, hs->max_year);
+		line << "\n";
+
+		uint8_t size = 0;
+		if ((hs->building_flags & TILE_SIZE_1x1) != 0) size = 0x11;
+		if ((hs->building_flags & TILE_SIZE_2x1) != 0) size = 0x21;
+		if ((hs->building_flags & TILE_SIZE_1x2) != 0) size = 0x12;
+		if ((hs->building_flags & TILE_SIZE_2x2) != 0) size = 0x22;
+		SetDParam(0, GB(size, 0, 4));
+		SetDParam(1, GB(size, 4, 4));
+		line << GetString(STR_HOUSE_PICKER_SIZE);
+		line << "\n";
+
+		auto str = BuildCargoAcceptanceString(GetAcceptedCargoOfHouse(hs), STR_HOUSE_PICKER_CARGO_ACCEPTED);
```

Comment:
`auto str` is a pretty generic name in a function dealing with strings. 😉 
```suggestion
		auto cargo_string = BuildCargoAcceptanceString(GetAcceptedCargoOfHouse(hs), STR_HOUSE_PICKER_CARGO_ACCEPTED);
```